### ISSUE TITLE
Error while indexing: "Too many open files"

### ISSUE THREAD

**Issue by eeickmeyer** [jflesch/paperwork#469]
- How to reproduce the problem? Import a folder containing PDF files.
- What is the expected behavior? OCRs files and indexes correctly.
- What behavior did you get? Stopped after displaying one file after import. Restarted Paperwork and exception was thrown during indexing process.
- What version of Paperwork are you using ? stable
- When run in a terminal, is there any uncatched Python exception raised ? See Below
- How did you install Paperwork ? (virtualenv / setup.py / distribution package) Readme instructions using pip
- What GNU/Linux distribution are you using? xubuntu

Terminal output below:

```
ERROR  paperwork.frontend.util.jobs   ===> Job IndexUpdater:0 raised an exception: <type 'exceptions.IOError'>: [Errno 24] Too many open files: '/home/erich/papers/20160408_1838_33_90/doc.pdf'
ERROR  paperwork.frontend.util.jobs    0: /usr/local/lib/python2.7/dist-packages/paperwork/frontend/util/jobs.py: L  178: _run
ERROR  paperwork.frontend.util.jobs    1: /usr/local/lib/python2.7/dist-packages/paperwork/frontend/mainwindow/__init__.py: L  405: do
ERROR  paperwork.frontend.util.jobs    2: /usr/local/lib/python2.7/dist-packages/paperwork/backend/docsearch.py: L  270: add_doc
ERROR  paperwork.frontend.util.jobs    3: /usr/local/lib/python2.7/dist-packages/paperwork/backend/docsearch.py: L  235: _update_doc_in_index
ERROR  paperwork.frontend.util.jobs    4: /usr/local/lib/python2.7/dist-packages/paperwork/backend/pdf/doc.py: L  195: get_docfilehash
ERROR  paperwork.frontend.util.jobs    5: /usr/local/lib/python2.7/dist-packages/paperwork/backend/common/doc.py: L  393: hash_file
ERROR  paperwork.frontend.util.jobs   ---> Job IndexUpdater:0 was started by:
ERROR  paperwork.frontend.util.jobs    0: /usr/local/bin/paperwork: L    6: <module>
ERROR  paperwork.frontend.util.jobs    1: /usr/local/lib/python2.7/dist-packages/paperwork/paperwork.py: L  114: main
ERROR  paperwork.frontend.util.jobs    2: /usr/local/lib/python2.7/dist-packages/paperwork/frontend/mainwindow/__init__.py: L 1855: __on_doc_exam_end
ERROR  paperwork.frontend.util.jobs    3: /usr/local/lib/python2.7/dist-packages/paperwork/frontend/util/jobs.py: L  247: schedule
```

Not sure if this is a bug or a limitation. I have many hundreds or thousands of PDFs I'm trying to import and index.

**Comment by tYYGH** [comment]
I already had this problem in the past: #405.
In the end, I created a file named ```/etc/security/limits.d/00-tYYGH.conf``` with this contents:
```
* soft nofile 8192
* hard nofile 8192
```

**Comment by eeickmeyer** [comment]
That workaround did not work for me as this isn't related to OCR, but in indexing the text created by the OCR. I did try it, but unfortunately it did not work. I will update with about how many PDF files I'm working with as an edit or in a further reply as I'm attempting to re-import and OCR all files after deleting the old index as a workaround. I tried this process earlier and it crashed, but in that scenario I didn't delete the old index.

**Comment by eeickmeyer** [comment]
Upon attempting the above method, I ran into this error:

'Error opening data file /usr/share/tesseract-ocr/tessdata/eng.traineddata
Please make sure the TESSDATA_PREFIX environment variable is set to the parent directory of your "tessdata" directory.
Failed loading language 'eng'
Tesseract couldn't load any languages!
INFO   paperwork.frontend.mainwindow.scan Animator: Angle 0: (20, 0) (678, 873) -> (419, 10) (658, 853)
Segmentation fault (core dumped)
'

Not sure if this should be a separate bug report or not.

**Comment by jflesch** [comment]
How many documents do you have ? (```ls -1 ~/papers | wc -l```)
And how many PDFs ? (```find ~/papers -name \*.pdf | wc -l)

**Comment by eeickmeyer** [comment]
1171, all PDFs.

**Comment by jflesch** [comment]
Will be fixed in 0.4:
- https://github.com/jflesch/paperwork-backend/commit/ca25c51fbca2acedfa294dad18129a6773ba970b
- https://github.com/jflesch/paperwork-backend/commit/b9def15b54ce31c69580e34bd604754a9a848af6
- db0a4dda35a2612e6c2c650af3f12f0f3511d54d
- 39fe95c06efc5e8d0a41ad29a2bc48a400ee2801

Sorry for the delay.
